pkg/compiler/codegen.go

```diff
@@ -870,6 +873,10 @@ func (c *codegen) Visit(node ast.Node) ast.Visitor {
 			ast.Walk(c, n.Fun)
 			emit.Opcodes(c.prog.BinWriter, opcode.CALLA)
 		case isSyscall(f):
+			if f.name == "Notify" {
+				tv := c.typeAndValueOf(n.Args[0])
+				c.emittedEvents[tv.Value.String()] = struct{}{}
```

Comment:
This one is weak in that only names are checked.

Reply:
Fixed.